samples/go/noms-merge/main_test.go

```diff
@@ -78,4 +79,41 @@ func (s *testSuite) TestWin() {
 	}
 }
 
-// TODO failure tests
+func (s *testSuite) TestLose() {
+	sp := spec.CreateDatabaseSpecString("ldb", s.LdbDir)
+	p, l, r := "parent", "left", "right"
+	type c struct {
+		args []string
+		err  string
+	}
+	cases := []c{
+		{[]string{"foo"}, "Incorrect number of arguments\n"},
+		{[]string{"foo", "bar"}, "Incorrect number of arguments\n"},
+		{[]string{"foo", "bar", "baz", "quux"}, "Incorrect number of arguments\n"},
+		{[]string{"foo", "bar", "baz"}, "--parent is currently required\n"},
+		{[]string{"--parent=" + p, sp, l + "!!", r}, "Invalid dataset " + l + "!!, must match [a-zA-Z0-9\\-_/]+\n"},
+		{[]string{"--parent=" + p, sp, l + "2", r}, "Dataset " + l + "2 has no data\n"},
+		{[]string{"--parent=" + p + "2", sp, l, r}, "Dataset " + p + "2 has no data\n"},
+		{[]string{"--parent=" + p, sp, l, r + "2"}, "Dataset " + r + "2 has no data\n"},
+		{[]string{"--parent=" + p, "--out-ds-name", "!invalid", sp, l, r}, "Invalid dataset !invalid, must match [a-zA-Z0-9\\-_/]+\n"},
+	}
+
+	db, _ := spec.GetDatabase(sp)
+	prep := func(dsName string) {
+		ds := dataset.NewDataset(db, dsName)
+		ds.CommitValue(types.NewMap(types.String("foo"), types.String("bar")))
+	}
+	prep(p)
+	prep(l)
+	prep(r)
+	db.Close()
+
+	for _, c := range cases {
+		var mainErr error
+		stdout, _ := s.Run(func() { mainErr = nomsMerge() }, c.args)
+		s.Empty(stdout, "Expected empty stdout for case: %#v", c.args)
+		if s.Error(mainErr) {
+			s.Equal(c.err, mainErr.Error(), "Unexpected output for case: %#v\n", c.args)
+		}
```

Comment:
this will fail if there isn't an error. `s.Error()` will fail, causing the test case to fail